src/Model/Comment.php

```diff
@@ -74,6 +74,11 @@ public function __construct( \WP_Comment $comment ) {
 	 */
 	protected function is_private() {
 
+		// A comment is considered private if it is attached to a private post.
+		if ( ! empty( $this->data->comment_post_ID ) && ( new Post( get_post( $this->data->comment_post_ID ) ) )->is_private() ) {
```

Comment:
I think we need to actually change this line. 

if the Post a comment was added to was deleted, then `get_post( $this-data->comment_post_ID )` will return null, and we can't instantiate `new Post( null )`, so we need to make sure `get_post()` returns a `WP_Post` before instantiating `new Post()`


Something like: 

```
// Comments not associated with a post ID should be considered private, we can return true here
if ( empty( $this->data->comment_post_ID ) {
  return true;
}

$commented_on = get_post( $this->data->comment_post_ID );

// If the commented on post doesn't exist, or the Post Model is private, this comment should be private too
if ( empty( $commented_on ) || new Post( get_post( $this->data->comment_post_ID ) ) )->is_private() ) {
  return true;
}
```

Reply:
After further testing it looks like WordPress is preventing orphaned comments when posts are deleted. As long as users are using WordPress in a standard way this is a case we should not need to worry about.